File: pkg/data/models/products/details/housing.go
package details

import (
	"BetterPC_2.0/pkg/data/models/products/general"
	"go.mongodb.org/mongo-driver/bson/primitive"
	"strconv"
)

type Housing struct {
	ID              primitive.ObjectID `bson:"_id,omitempty"`
	General         general.General    `bson:"general"`
	FormFactor      string             `bson:"form_factor"`
	DriveBays       driveBays          `bson:"drive_bays"`
	MbFormFactor    string             `bson:"mb_form_factor"`
	PsFormFactor    string             `bson:"ps_form_factor"`
	ExpansionSlots  int                `bson:"expansion_slots"`
	GraphicCardSize int                `bson:"graphic_card_size"`
	CoolerHeight    int                `bson:"cooler_height"`
	Size            []int              `bson:"size"`
	Weight          float64            `bson:"weight"`
}

type driveBays struct {
	D35 int `bson:"3_5"`
	D25 int `bson:"2_5"`
}

type UpdateHousingInput struct {
	General         *general.General `bson:"general"`
	FormFactor      *string          `bson:"form_factor"`
	DriveBays       *driveBays       `bson:"drive_bays"`
	MbFormFactor    *string          `bson:"mb_form_factor"`
	PsFormFactor    *string          `bson:"ps_form_factor"`
	ExpansionSlots  *int             `bson:"expansion_slots"`
	GraphicCardSize *int             `bson:"graphic_card_size"`
	CoolerHeight    *int             `bson:"cooler_height"`
	Size            *[]int           `bson:"size"`
	Weight          *float64         `bson:"weight"`
}

func (housing Housing) GetProductModel() string {
	return housing.General.Model
}

func (housing Housing) Standardize() general.StandardizedProductData {
	var product general.StandardizedProductData
	product.ProductHeader.ID = housing.ID.Hex()
	product.ProductHeader.ProductType = "housing"
	product.Name = housing.General.Model
	product.General = housing.General
	product.Description = "Form Factor: " + housing.FormFactor + ", Motherboard Form Factor: " +
		housing.MbFormFactor + ", Expansion Slots: " + strconv.Itoa(housing.ExpansionSlots)
	return product
}

func (housing Housing) ProductFinalPrice() int {
	return housing.General.CalculateFinalPrice()
}

func (h UpdateHousingInput) Validate() error {
	return general.ValidateStruct(&h)
}

File: frontend/better_pc/src/modules/products/models/housing.ts
import {BaseProduct} from "@/modules/products/models/product_common"; // Общий интерфейс General

export interface DriveBays {
    d35: number; // Количество отсеков для 3.5" накопителей
    d25: number; // Количество отсеков для 2.5" накопителей
}

export interface Housing extends BaseProduct {
    formFactor: string; // Форм-фактор корпуса (например, "ATX", "mATX", "ITX")
    driveBays: DriveBays; // Объект с информацией о количестве отсеков для накопителей
    mbFormFactor: string; // Поддерживаемые форм-факторы материнских плат
    psFormFactor: string; // Форм-фактор блока питания (например, "ATX", "SFX")
    expansionSlots: number; // Количество слотов расширения
    graphicCardSize: number; // Максимальная длина видеокарты (в мм)
    coolerHeight: number; // Максимальная высота процессорного кулера (в мм)
    size: number[]; // Размеры корпуса (длина, ширина, высота, в мм)
    weight: number; // Вес корпуса (в кг)
}
